### PR TITLE
Added companion ad positioning also fix ad presentation inside player…

### PR DESCRIPTION
**Pull request by sambua**
… after new changes
added positions related for the player, after dimensions for the player `[heigh, width]|position`
There also existed selector from the first changes where container could be in any part of the page.

### CODE REVIEW

File: src/ads/ima/requester.js [hunk 1]
```diff
@@ -406,40 +406,80 @@ Scoped.define("module:Ads.IMARequester", [
...
-             * @param options
+             * @param dyn
              */
-            _showCompanionAd: function(ad, options) {
+            _showCompanionAd: function(ad, dyn) {
+                var element, playerElement, position, selector, options, height, width;
+                playerElement = dyn.activeElement();
+                options = dyn.get("companion-ad");
+                if (options.split('|').length > 0) {
+                    position = options.split('|')[1] || 'bottom';
```
Comment (ntshcalleia) [inline src/ads/ima/requester.js:416]: Hi, @sambua! Wouldn't it be better to add a new attribute `companion-ad-position` on the player with `bottom` as default value? It just seems more self explanatory than having to add `|` and a value in the `companion-ad` attribute.

File: src/ads/ima/requester.js [hunk 2]
```diff
@@ -406,40 +406,80 @@ Scoped.define("module:Ads.IMARequester", [
...
-                var selector = options[0];
-                if (!selector) return;
-                var element = document.getElementById(selector);
+                selector = options[0];
+                if (selector) {
+                    element = document.getElementById(selector);
+                } else {
+                    element = document.createElement('div');
+                }
                 if (!element) return;
```
Comment (ntshcalleia) [inline src/ads/ima/requester.js:425]: This should create element as fallback when there's no object with id `selector`. 🙂
```suggestion
                if (selector) element = document.getElementById(selector);
                if (!element) element = document.createElement('div');
```

File: src/ads/ima/requester.js [hunk 3]
```diff
@@ -406,40 +406,80 @@ Scoped.define("module:Ads.IMARequester", [
...
-                var dimensions = options[1];
-                if (!dimensions) return;
+                var dimensions = options[1].split(',');
+                var isFluid = dimensions === 'fluid';
```
Comment (ntshcalleia) [inline src/ads/ima/requester.js:427]: dimensions is an array, so we need to do this here
```suggestion
                var isFluid = dimensions[0] === 'fluid';
```

**Review by ntshcalleia — CHANGES_REQUESTED** [review]
Hi, @sambua!

As mentioned on Slack, width and height don't seem to be affecting the dimensions of the companion-ad. I also added a few comments, when you have the time please take a look at them. 🙂

**Comment by sambua** [comment]
Thanks @ntshcalleia for the feedback. 

1. For me especially preferable use one attribute not use additional for small .
2. Regarding fallback, why I'm not using that option, because selector should be exact. In other case it can break design. Like in youtube they are showing companion ad, in the sidebar, and if it will show under the player surely it will break the UI.
3. Regarding fluid it's correct, I've replaced dimensions position in new changes, by adding slip function before fluid checking. Will fix it thanks!

And dimensions are required parameter per VAST documentation, and IMA SDK require it to be able show them.

**Comment by sambua** [comment]
@ntshcalleia I've also just tested, with empty `[]` it will take video dimensions and show closer to that. And pushed latest changes

**Review by ntshcalleia — APPROVED** [review]

**Comment by ntshcalleia** [comment]
Got it! Thanks for replying, @sambua!
